create_derivatives/routines.py

```diff
@@ -78,8 +79,30 @@ class ManifestMaker:
 
 
 class AWSUpload:
-    # TO DO: upload files and PDFs
-    pass
+
+    def __init__(self):
+        self.aws_client = AWSClient(*settings.AWS)
+
+    def run(self, replace):
+        uploaded_bags = []
+        for bag in Bag.objects.filter(process_status=Bag.MANIFESTS_CREATED):
+            pdf_dir = join(bag.bag_path, 'data', 'PDF')
```

Comment:
I feel like lines 89-94 could be taken care of in a for loop.

Reply:
I'm also not convinced it makes sense to call `check_dir_exists` here. Are we trying to produce a specific Exception message? Can't you just try to upload stuff and handle the exception at a higher level?